temporal-sdk/src/main/java/io/temporal/common/converter/DataConverter.java

```diff
@@ -174,7 +176,10 @@ default Object[] fromPayloads(
    * @throws NullPointerException if failure is null
    */
   @Nonnull
-  TemporalFailure failureToException(@Nonnull Failure failure);
+  default TemporalFailure failureToException(@Nonnull Failure failure) {
+    Preconditions.checkNotNull(failure, "failure");
+    return new DefaultFailureConverter().failureToException(failure, this);
```

Comment:
I didn't make this a interface variable because then it would be public

Reply:
Could make it a singleton inside that class, e.g.

```suggestion
return DefaultFailureConverter.getInstance().failureToException(failure, this);
```

though I suspect instantiating this for every failure is mostly harmless and you couldn't make the constructor private or protected anyways for other compatibility reasons (so it'd be a singleton w/ a public constructor which is kinda weird).